lib/reindex/s3_bucketd.py

```diff
@@ -48,6 +48,14 @@ def inner(*args, **kwargs):
         return urllib.parse.quote(val.encode('utf-8'))
     return inner
 
+def as_set(func):
```

Comment:
I recommend using functools.wraps for all decorators as it handles all the extra bits plain decorators do not.
https://docs.python.org/3.8/library/functools.html#functools.wraps

Reply:
I've removed the decorator per your other comment, but what are the extra bits `wraps` preserves? Docs seem to indicate the docstring and function name are saved, is it only metadata or have I been avoiding a massive footgun by chance all these years?

Reply:
I think its just metadata like docstring and name, I think it does impact special cases like open API spec tools or sqlalchemy will sometimes use that data. It might cause an issue if you stack decorators and they aren't wrapped, then the metadata gets real fuzzy. But it is a relatively small footgun at most, I mention it mostly as a safety thing. And not everyone knows about functools ( itertools is also great ).